### PR TITLE
Qt6 for macOS w/ universal builds

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by Beerosagos** [comment]
Also, I noticed a lot of this errors in the console at the end of the compiling. It still works, but wanted to let you know about it:
```
cd build/osx/ && macdeployqt BitBox.app
ERROR: Could not parse otool output line: "/Users/beerosagos/repo/beero/bitbox-wallet-app/frontends/qt/build/osx/BitBox.app/Contents/MacOS/BitBox (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtWebEngineWidgets.framework/Versions/A/QtWebEngineWidgets (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtPrintSupport.framework/Versions/A/QtPrintSupport (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtWidgets.framework/Versions/A/QtWidgets (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtWebEngineCore.framework/Versions/A/QtWebEngineCore (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtQuick.framework/Versions/A/QtQuick (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtOpenGL.framework/Versions/A/QtOpenGL (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtGui.framework/Versions/A/QtGui (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtQmlModels.framework/Versions/A/QtQmlModels (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtWebChannel.framework/Versions/A/QtWebChannel (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtQml.framework/Versions/A/QtQml (architecture arm64):"
ERROR: Could not parse otool output line: "/Users/beerosagos/Qt/6.2.4/macos/lib/QtNetwork.framework/Versions/A/QtNetwork (architecture arm64):"
[...]
```

File: docs/BUILD.md [hunk 1]
```diff
@@ -30,17 +30,28 @@ Build artifacts:
...
+aqt list-qt mac desktop --arch 6.2.4
+aqt install-qt mac desktop 6.2.4 --modules qtpositioning qtserialport qtwebchannel qtwebengine --outputdir ~/Qt
+```
+
+Make sure you have `qt@6/bin`,  `qt@6/libexec`, `go@1.23/bin` and `go/bin` in your PATH, i.e. add to your `.zshrc`:
 
 ```bash
-export PATH="$PATH:/usr/local/opt/qt@5/bin"
+export PATH="$PATH:~/Qt/6.2.4/macos/bin"
+export PATH="$PATH:~/Qt/6.2.4/macos/libexec"
```
[the remark this answers is not in the document]
Comment (benma) [reply docs/BUILD.md:50]: Why?

File: README.md [hunk 2]
```diff
@@ -53,14 +53,6 @@ To build the app or run the development workflow, the following dependencies nee
 - [Qt5](https://www.qt.io) version 5.15.2
   - install Qt for your platform, including the WebEngine component
 
-#### Environment variables
-
-Make sure the environment variables for qt, go and node are set, see also [docs/BUILD.md](./docs/BUILD.md) for platform specific instructions
-
-To initialize the build environment and install the required go utilities (linters, ...), call
-* `make envinit`, or
```
[the remark this answers is not in the document]
Comment (benma) [reply README.md:61]: That's not needed for building the macOS app.

**Comment by benma** [comment]
> ERROR: Could not parse otool output line

@Beerosagos interesting, and yeah it seems to work regardless. I can't find much info about this bug.

https://forum.qt.io/topic/136644/macqtdeploy-cmake-arm-m1-to-intel-oh-dear/3

> you can ignore the otool messages if I remember correctly. There has been a bug report on Qt for this and it is basically the script doesn't parse all output of arm64 otool correctly.

I wish they had a reference :smile:

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Beerosagos) [reply README.md:61]: No, but we'll still need this instruction for building in other environments, right? Maybe I am missing something

File: docs/BUILD.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Beerosagos) [reply docs/BUILD.md:50]: In my tests it didn't work with `~`

[the remark this answers is not in the document]
Comment (NickeZ) [reply docs/BUILD.md:50]: Tilde is not expanded properly.

[the remark this answers is not in the document]
Comment (NickeZ) [reply docs/BUILD.md:50]: https://apple.stackexchange.com/questions/255353/any-difference-between-and-home

**Review by Beerosagos — APPROVED** [review]
Nice! tACK

**Comment by benma** [comment]
Tested successfully on arm64 and amd64 machines.

**Pull request merged**